packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -147,6 +148,11 @@ export class ArmParameters {
 
 export const ProjectMigratorMW: Middleware = async (ctx: CoreHookContext, next: NextFunction) => {
   if ((await needMigrateToArmAndMultiEnv(ctx)) && checkMethod(ctx)) {
+    if (!checkUserTasks(ctx)) {
```

Comment:
What if moving this check before line 150? Just to save extra check time.

Reply:
@swatDong  You can see line 205, to avoid executing next middleware that will cause unrelated error.